### PR TITLE
Update seastar submodule and main

### PR DESCRIPTION
**Pull request by tchaikov**
this change also includes change to main, to make this commit compile.
see below:

* seastar 9b6e181e42...9cbc1fe889 (46):
  > Merge 'Make io-tester jobs share sched classes' from Pavel Emelyanov
  > io_tester.md: Update the `rps` configuration option description
  > io_tester: Add option to limit total number of requests sent
  > Merge 'Keep outgoing queue all cancellable while negotiating (again)' from Pavel Emelyanov
  > io_tester: Add option to share classes between jobs
  > rpc: Abort connection if send_entry() fails
  > Merge 'build: build dpdk with `-fPIC` if BUILD_SHARED_LIBS' from Kefu Chai
  > build: cooking.sh: use the same BUILD_SHARED_LIBS when building ingredients
  > build: cooking.sh: use the same generator when building ingredients
  > core/memory: handle `strerror_r` returning static string
  > Merge 'build, rpc: lz4 related cleanups' from Kefu Chai
  > build, rpc: do not support lz4 < 1.7.3
  > build: set the correct version when finding lz4
  > build: include CheckSymbolExists
  > rpc: do not include lz4.h in header
  > build: set CMP0135 for Cooking.cmake
  > docs: drop building-*.md
  > Merge 'seastar-addr2line: cleanups' from Kefu Chai
  > seastar-addr2line: refactor tests using unittest
  > seastar-addr2line: extract do_test() and main()
  > seastar-addr2line: do not import unused modules
  > scheduling: add a `rename` callback to scheduling_group_key_config
  > reactor: syscall thread: wakeup up reactor with finer granularity
  > build: build dpdk with `-fPIC` if BUILD_SHARED_LIBS
  > build: extract dpdk_extra_cflags out
  > core/sstring: remove a temporary variable
  > Merge 'treewide: include what we use, and add a checkheaders target' from Kefu Chai
  > perftune.py: auto-select the same number of IRQ cores on each NUMA
  > prometheus: remove unused headers
  > core/sstring: define <=> operator for sstring
  > Merge 'core: s/reserve_additional_memory/reserve_additional_memory_per_shard/' from Kefu Chai
  > include: do not include <concepts> directly
  > coding_style: note on self-contained header requirement
  > circileci: build checkheaders in addition to default target
  > build: add checkheaders target
  > net/toeplitz: s/u_int/unsigned/
  > net/tcp-stack: add forward declaration for seastar::socket
  > core, net, util: include used headers

* main: set reserved memory for wasm on per-shard basis

  this change is a follow-up of
  f05d612da8854b9bc4be0a2cf03f906bccb76330 and
  4a0134a097fad3144f7b7d005d92cd87a08d6576.

  this change depends on the related change in Seastar to reserve
  additional memory on a per-shard basis.

  per Wojciech Mitros's comment:

  > it should have probably been 50MB per shard

  in other words, as we always execute the same set of udf on all
  shards. and since one cannot predict the number of shards, but she
  could have a rough estimation on the size of memory a regular (set
  of) udf could use. so a per-shard setting makes more sense.

Signed-off-by: Kefu Chai <kefu.chai@scylladb.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
@avikivity could you include this change when you queue a change to bump up seastart submodule? as this change depends on https://github.com/scylladb/seastar/commit/737a43a26a3853536262ae08b8de10c4a27ce31c , which is not included by the seastar submodule of scylladb master HEAD at this moment.

**Comment by tchaikov** [comment]
@wmitros @avikivity Wojciech and Avi, could you help review this change?

**Comment by tchaikov** [comment]
the change of `core/sstring: define <=> operator for sstring` in seastar should help to silence warnings like

```
12:51:39  ./data_dictionary/storage_options.hh:26:21: error: explicitly defaulted three-way comparison operator is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
12:51:39          friend auto operator<=>(const s3&, const s3&) = default;
12:51:39                      ^
```

**Review by wmitros — COMMENTED** [review]
The adjustments to the `wasm_udf_reserved_memory` look good, the rest of the wasm code already assumed we reserve given memory per shard, so it shouldn't require any more changes.

I'm not aware about any requirements about updating the seastar module, so I can't really comment on that

**Comment by avikivity** [comment]
The changelog says

    99b6e181e4...9cbc1fe889

but git diff says

--- a/seastar
+++ b/seastar
@@ -1 +1 @@
-Subproject commit 9b6e181e424fd18b780ade4df59e2c0c3a806742
+Subproject commit 9cbc1fe8898ab0e5f0a7a83f5932ee0be00acf8d

So something was wrong when generating the patch. When you create it (and the changelog) the submodule should be synchronized to master at the time.

**Comment by tchaikov** [comment]
@avikivity i am sorry. in future, will sync the submodule when updating it. updated the commit message and repushed.
